File: OneQuestionPerDay/104/maximum_depth_of_binary_tree3.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        level = 0
        queue = [root]

        while queue:
            n = len(queue)
            level += 1

            for i in range(n):
                if queue[i].left is not None:
                    queue.append(queue[i].left)
                if queue[i].right is not None:
                    queue.append(queue[i].right)

            queue = queue[n:]
        return level

File: OneQuestionPerDay/104/maximum_depth_of_binary_tree3.go
package main

func maxDepth3(root *TreeNode) int {
	if root == nil {
		return 0
	}
	stack := []*TreeNode{root}
	level := 0

	for len(stack) > 0 {
		num := len(stack)
		for i := 0; i < num; i++ {
			if stack[i].Left != nil {
				stack = append(stack, stack[i].Left)
			}
			if stack[i].Right != nil {
				stack = append(stack, stack[i].Right)
			}
		}
		stack = stack[num:]
		level++
	}
	return level
}
